src/storage/storage_structure/storage_structure.cpp

```diff
@@ -139,17 +207,12 @@ void BaseColumnOrList::readNodeIDsBySequentialCopyWithSelState(Transaction* tran
 void BaseColumnOrList::readNodeIDsFromAPageBySequentialCopy(Transaction* transaction,
     const shared_ptr<ValueVector>& vector, uint64_t vectorStartPos, page_idx_t physicalPageIdx,
     uint16_t pagePosOfFirstElement, uint64_t numValuesToRead,
-    NodeIDCompressionScheme& nodeIDCompressionScheme, bool isAdjLists) {
+    NodeIDCompressionScheme& nodeIDCompressionScheme, const set_vec_null_func& setVecNullFunc) {
```

Comment:
I might prefer to use bool instead of a function pointer.

For bool, the branch prediction will always succeed, so we won't have actual performance penalty.

For function pointer, my understanding is that it will always trigger a virtual table lookup which could an overhead. Correct me if I'm wrong. 

I agree the bool name should change to hasNoNullGuarantee and maybe comment that for adjList we don't need to set null bits for each entry.

Reply:
Changed to `bool hasNoNullGuarantee`